### ISSUE TITLE
Problem with k.function

### ISSUE THREAD

**Issue by mijosch** [plaidml/plaidml#297]
```
---------------------------------------------------------------------------
Unknown                                   Traceback (most recent call last)
<ipython-input-44-ecbdf7bf06ac> in <module>()
      4 grads = k.gradients(cat0_Output,last_conv_layer.output)[0]
      5 pooled_grads = k.mean(grads,axis=(0,1,2))
----> 6 iterate = k.function([model.input],[pooled_grads,last_conv_layer.output[0]])
      7 pooled_grads_value, conv_layer_output_value = iterate([im])

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\keras\backend.py in function(inputs, outputs, updates, name)
    684     if name == None:
    685         name = ''
--> 686     return _Function(inputs, outputs, updates, name)
    687 
    688 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\keras\backend.py in __init__(self, inputs, outputs, updates, name)
    146             list(zip(self._output_names, outputs)),
    147             updates,
--> 148             name=name)
    149         self._invoker = plaidml.Invoker(_ctx, self._func)
    150 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\tile.py in compose(ctx, dev, inputs, outputs, updates, name)
   1060         for (output_name, val) in outputs:
   1061             binding = val.bind(bindings)
-> 1062             composer.add_output(output_name, binding)
   1063         for (original, updated) in updates:
   1064             composer.add_update(original.bind(bindings), updated.bind(bindings))

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\__init__.py in add_output(self, name, val)
   1383 
   1384     def add_output(self, name, val):
-> 1385         _lib().plaidml_add_composer_output(self, name.encode(), val)
   1386 
   1387     def add_dependency(self, applier):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\__init__.py in _check_err(self, result, func, args)
    744         if func == self.plaidml_map_buffer_current and args[2]:
    745             return None
--> 746         self.raise_last_status()
    747 
    748 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\library.py in raise_last_status(self)
    129 
    130     def raise_last_status(self):
--> 131         raise self.last_status()
    132 
    133     def _logger_callback(self, unused_arg, level, msg):

Unknown: Binding missing placeholder
```

**Comment by mijosch** [comment]
any idea?

**Comment by iperov** [comment]
provide full code

**Comment by mijosch** [comment]
I want to make an activation map for a prediction like described in a book

this works with tensorflow backend
```
pretrained_model = applications.vgg16.VGG16(weights='D:\\Keras_Weigths\\vgg16_weights_tf_dim_ordering_tf_kernels.h5',include_top=True)
cimage = cv2.imread("D:\\Bilder\\KI\\3autos.jpg")
cimage = cv2.resize(cimage,(224,224))
x = np.expand_dims(cimage, axis=0)
not_used=pretrained_model.predict(x)
pred_out = pretrained_model.output[:,436]
last_conv_layer = pretrained_model.get_layer('block5_conv3')
grads = k.gradients(pred_out,last_conv_layer.output)[0]
pooled_grads = k.mean(grads,axis=(0,1,2))
iterate = k.function([pretrained_model.input],[pooled_grads,last_conv_layer.output[0]])
pooled_grads_value, conv_layer_output_value = iterate([x])
```
>>Unknown: More indexes than dimensions for tensor: 1:X_T194 1 > 0



another problem which also works with tensorflow backend
this does not work also
```

#agmax = np.argmax(not_used)
#pred_out = pretrained_model.output[:,agmax]

```
>> AttributeError: 'numpy.int64' object has no attribute 'start'

here the error seems to be in the tile.py
```

~\AppData\Local\Continuum\anaconda3\lib\site-packages\plaidml\tile.py in _parse_slice(dims, key, idx)
    483                 val, val.__class__.__name__))
    484 
--> 485         check(key[idx].start)
    486         check(key[idx].stop)
    487         check(key[idx].step)
```

**Comment by mijosch** [comment]
problem 2 solved by myself
plaidml cannot handle numpy types

```
agmax = np.argmax(not_used)
pred_out = pretrained_model.output[:,agmax.item()]
```
>> fixed the issue

**Comment by brianretford** [comment]
Still blocked on issue 1?

**Comment by mijosch** [comment]
Yes and i cannot figure out the problem,
seems like the code is a common used pattern with tensorflow to make a weighted activation map

**Comment by brianretford** [comment]
Yeah usually these kind of bugs are something dumb in our keras backend. @tzerrell could you take a look?

**Comment by mijosch** [comment]
Some news to this issue?

**Comment by tzerrell** [comment]
This looks like probably a duplicate of #296
